src/main/java/uk/gov/pay/adminusers/service/ServiceRoleCreator.java

```diff
@@ -34,20 +34,20 @@ public ServiceRoleCreator(UserDao userDao, ServiceDao serviceDao, RoleDao roleDa
     }
 
     @Transactional
-    public Optional<User> doCreate(String userExternalId, String serviceExternalId, String roleName) {
+    public Optional<User> doCreate(String userExternalId, String serviceExternalId, RoleName roleName) {
         Optional<UserEntity> userMaybe = userDao.findByExternalId(userExternalId);
-        if (!userMaybe.isPresent()) {
+        if (userMaybe.isEmpty()) {
             return Optional.empty();
         }
 
         Optional<ServiceEntity> serviceMaybe = serviceDao.findByExternalId(serviceExternalId);
-        if (!serviceMaybe.isPresent()) {
+        if (serviceMaybe.isEmpty()) {
             throw serviceDoesNotExistError(serviceExternalId);
```

Comment:
I'm not used to seeing exceptions as static references, I would always expect them to be a class, based on what we see in connector